ecoscope/base/base.py

```diff
@@ -516,6 +518,165 @@ def turn_angle(traj):
 
         return angles.rename("turn_angle").reindex(self.index)
 
+    def upsample(self, upsample_time):
+        """
+        Function to upsample a trajectory. The upsampling works by doing a linear
+        interpolation between each of the relocation points.
+
+        Parameters
+        ----------
+        up_sample_time: int
+            Upsampling interval (in seconds)
+        Returns
+        -------
+        ecoscope.base.Relocations
+        """
+
+        upsampled_relocs = []
+
+        def compute(traj_ind):
+            subject_name = traj_ind.name
+            traj_ind.crs = self.crs
+            utm_crs = traj_ind.estimate_utm_crs()
+            traj_ind.to_crs(utm_crs, inplace=True)
+            logging.info(f"Resampling relocations for subject {subject_name}")
+
+            traj_ind = traj_ind.sort_values("segment_start")
+            times = pd.date_range(
+                start=traj_ind.iloc[0]["segment_start"],
+                end=traj_ind.iloc[-1]["segment_end"],
+                freq=pd.Timedelta(seconds=upsample_time),
+            )
+            index = traj_ind.segment_start.searchsorted(
```

Comment:
What about resampled times that don't fall within a `Trajectory` segment (multiple collars)? I think you might need a second searchsorted along with filtering.

Reply:
The function works with resampled times that don't fall within a trajectory segment (see tests for an example). I also think it can handle big gaps in the data, such as re-collaring ones. I haven't test it properly tho but I can make a synthetic examples to see if it works

Reply:
Okay, fixed now